mitodl/mitxpro
# feat: add Global Alumni in external course sync

### What are the relevant tickets?
https://github.com/mitodl/hq/issues/6107

### Description (What does it do?)
This PR refactors the external course sync code to make it more generic and adds `Global Alumni` as a vendor. Additionally, it introduces the `sync_daily` property in the Platform model, which is set to `False` by default. For platforms that have an external API and require daily synchronization, the `sync_daily` property should be set to `True`. This can be updated via the Django admin interface.

### How can this be tested?
- Run the migrations using `./manage.py migrate`
- Follow the testing steps in https://github.com/mitodl/mitxpro/pull/2998 to test the management command `sync_external_course_runs`
- To test the `task_sync_external_course_runs`, [go to platform model in django admin](http://localhost:8053/admin/courses/platform/)
- Create "Emeritus" and "Global Alumni" platforms if not already exist.
- Open django shell using `./manage.py shell`
- Run the following code:
   ```
   In [1]: from courses.tasks import task_sync_external_course_runs
   In [2]: task_sync_external_course_runs.delay()
   Out[2]: <AsyncResult: d6521ec0-8ba7-4852-af18-e38febdae33e>   
   ``` 
- Monitor the logs of the celery task, the task should be successful and it should create or update the required data using the API response.

### Additional Context
The Emeritus and Global Alumni APIs return data with the same fields, except for the CEU field, which is present in Emeritus but not in Global Alumni.
